common/functions/src/scalars/hashes/sha1hash.rs

```diff
@@ -12,80 +12,25 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::fmt;
-
-use common_datavalues::columns::DataColumn;
 use common_datavalues::prelude::*;
-use common_datavalues::DataSchema;
-use common_datavalues::DataType;
-use common_exception::ErrorCode;
-use common_exception::Result;
 
-use crate::scalars::function_factory::FunctionDescription;
-use crate::scalars::function_factory::FunctionFeatures;
-use crate::scalars::Function;
+use crate::scalars::strings::String2StringFunction;
+use crate::scalars::strings::StringOperator;
 
-#[derive(Clone)]
-pub struct Sha1HashFunction {
-    display_name: String,
-}
+#[derive(Clone, Default)]
+pub struct Sha1 {}
 
-impl Sha1HashFunction {
-    pub fn try_create(display_name: &str) -> Result<Box<dyn Function>> {
-        Ok(Box::new(Sha1HashFunction {
-            display_name: display_name.to_string(),
-        }))
+impl StringOperator for Sha1 {
+    #[inline]
+    fn apply_with_no_null<'a>(&'a mut self, s: &'a [u8], buffer: &mut [u8]) -> usize {
+        // TODO sha1 lib doesn't allow encode into buffer...
+        buffer.copy_from_slice(sha1::Sha1::from(s).digest().to_string().as_ref());
```

Comment:
https://docs.rs/sha1/0.6.0/src/sha1/lib.rs.html#187 
```
impl Digest {
    /// Returns the 160 bit (20 byte) digest as a byte array.
    pub fn bytes(&self) -> [u8; DIGEST_LENGTH] {
```

Reply:
Or can we use `write!`, ` std::io::Write::write_fmt` to reduce the String allocation?

Reply:
function should return hex instead of raw bytes - hex is bytes x 2
thee problem was in multi value case 
I forgot to make slice of whole slice (`copy_from_slice` reqs slices of same length)

i think `hex::encode_to_slice` works better (hex crate already in our Cargo)